chat-plugins/scavengers.js

```diff
@@ -144,13 +199,86 @@ exports.commands = {
+		}
+		buffer += "</table></div>";
+		return this.sendReply(buffer);
+	},
+	scavaddpoints: 'scavengersaddpoints',
+	scavengersaddpoints: function (target, room, user) {
+		if (room.id !== 'scavengers') return this.errorReply('This command can only be used in the Scavengers room.');
+		if (!this.can('mute', null, room)) return false;
+		let split = target.split(",");
+		if (split.length !== 2) return this.errorReply("You must specify a username and the number of points.");
+		let numPoints = parseInt(split[1]);
+		if (!numPoints || numPoints < 1) return this.errorReply("Invalid number of points.");
+		addPoints(split[0], numPoints);
+		scavWriteData();
+		return this.sendReply(`${numPoints} points awarded to ${scavengersLB[toId(split[0])].name}.`);
+	},
+	scavengersremovepoints: "scavrmpoints",
+	scavengersrmpoints: "scavrmpoints",
+	scavremovepoints: "scavrmpoints",
+	scavrmpoints: function (target, room, user) {
+		if (room.id !== 'scavengers') return this.errorReply('This command can only be used in the Scavengers room.');
+		if (!this.can('mute', null, room)) return;
+		let split = target.split(",");
+		if (split.length !== 2) return this.errorReply("You must specify a username and the number of points.");
+		let numPoints = parseInt(split[1]);
+		if (!numPoints || numPoints < 1) return this.errorReply("Invalid number of points.");
+		let id = toId(split[0]);
+		if (!(id in scavengersLB)) return this.errorReply("That user does not have any points.");
+		if (scavengersLB[id].points < numPoints) return this.errorReply("That user does not have enough points to remove.");
+		scavengersLB[id].points -= numPoints;
+		scavWriteData();
+		return this.sendReply(`${numPoints} points removed from ${scavengersLB[id].name}.`);
+	},
+
+	scavresetlb: function (target, room, user) {
+		if (room.id !== 'scavengers') return this.errorReply('This command can only be used in the Scavengers room.');
+		if (!this.can('declare', null, room)) return false;
+		scavengersLB = {};
+		scavWriteData();
+		return this.sendReply("The leaderboard has been reset.");
+	},
+
+	scavengersrank: 'scavrank',
+	scavrank: function (target, room, user) {
+		if (room.id !== 'scavengers') return this.errorReply('This command can only be used in the Scavengers room.');
+		let name = target;
+		if (!name) name = user.name;
+		let id = toId(name);
+		if (!(id in scavengersLB)) return this.sendReplyBox("User '" + name + "' does not have any points on the scavengers leaderboard");
+		let ladder = getLadder();
+		let i;
+		for (i = 0; i < ladder.length; i++) {
+			if (id === toId(ladder[i][1])) break;
+		}
+		if (i === ladder.length) { // shouldn't ever happen
+			return;
+		}
+		return this.sendReplyBox("User '" + ladder[i][1] + "' is #" + (i + 1) + " on the official leaderboard with " + ladder[i][0] + " points.");
```

Comment:
Probably template strings here, and in line 271 (and similar places) would be better.